owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,219 @@
+#[derive(Clone, Copy)]
+#[derive(Serialize,Deserialize)]
+enum ErrorCode {
+    IncompleteVoucher,
+    ParseError,
+    InvalidNumberOfVouchers,
+    UnwonedVoucher,
+    RequireContentTypeHeader,
+    RequireVoucherCountHeader,
+    NoOwnershipVoucherFound,
+    UnknownDevice,
+}
+
+#[derive(Serialize,Deserialize)]
+struct ErrorMsg<T:Serialize> {
+    error_code: Option<ErrorCode>,
+    error_details : Option<ErrorDetails<T>>
+}
+#[derive(Serialize,Deserialize)]
+struct ErrorDetails<T:Serialize >  {
+    key : Option<ErrorCode>,
+    value: Option<T>
+}
+
+trait Response<T> {
+    fn generate(e: ErrorCode,value: T ) -> Json;
+}
+
+impl<T> Response<T> for ErrorMsg<T> where T: Serialize {
+    fn generate(e: ErrorCode,value: T) -> Json {
+          let e=     ErrorMsg {
+                error_code:  Some(e),
+                error_details: Some(ErrorDetails{
+                    key : Some(e),
+                    value: Some(value),
+                })
+            };
+        warp::reply::json(&e)
+    }
+}
+
+
+
+async fn add_ov(
+    count: HeaderValue,
+    content_type: HeaderValue, 
+    body: impl Buf,
+    user_data: Arc<OwnerServiceUD>,
+    owner_key: PKey<Private>,
+) -> Result<impl warp::Reply,warp::Rejection> 
+{   
+    // check for ov type
+    let content_type = content_type.to_str().unwrap();
+    let ovs;
+    match content_type {
+        "application/x-pem-file" => ovs= OwnershipVoucher::many_from_pem(body.chunk()),
+        _ => {
+            let  e =  ErrorMsg::generate(ErrorCode::RequireContentTypeHeader, ACCEPTED_CONTENT_TYPE);
+            return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+        }
+    }
+
+    // parse-error
+    if ovs.is_err() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, ovs.unwrap_err().to_string());
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check-empty
+    let ovs = ovs.unwrap();
+    if ovs.is_empty() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, "Ownership voucher not found!!");
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check if X-Number-Of-Vouchers present
+    let count = count.to_str().unwrap().parse::<usize>();
+    if count.is_err() {
+        let  e =  ErrorMsg::generate(ErrorCode::RequireVoucherCountHeader, "X-Number-Of-Vouchers: Should be postive Interger");
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+    
+
+    // check if count matches
+    if ovs.len() != count.unwrap() {
+        let  e =  ErrorMsg::generate(ErrorCode::InvalidNumberOfVouchers, ovs.len());
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+    let mut owned_guid = Vec::new();
+    let mut unowned_guid = Vec::new();
+    for ov in ovs.iter() {
+        if validate_ov(ov,&owner_key){
+            let o = ov.header().guid().to_string();
+            owned_guid.push(o);
+            async {user_data
```

Comment:
This means that if there are some unowned vouchers in the set, the others are actually imported.